paasta_tools/vitesscluster_tools.py

```diff
@@ -15,6 +17,594 @@
+                            {
+                                "key": "yelp.com/pool",
+                                "operator": "In",
+                                "values": [paasta_pool],
+                            }
+                        ]
+                    }
+                ]
+            }
+        }
+    }
+    return spec
+
+
+def build_extra_labels(paasta_pool, paasta_cluster):
+    """
+    Build extra labels to adhere to paasta contract
+    """
+    extra_labels = {
+        "yelp.com/owner": "dre_mysql_working_hours",
+        "paasta.yelp.com/cluster": paasta_cluster,
+        "paasta.yelp.com/pool": paasta_pool,
+    }
+    return extra_labels
+
+
+def build_extra_env(paasta_cluster):
+    """
+    Build extra env to adhere to paasta contract
+    """
+    extra_env = [
+        {
+            "name": "PAASTA_POD_IP",
+            "valueFrom": {"fieldRef": {"fieldPath": "status.podIP"}},
+        },
+        {"name": "PAASTA_CLUSTER", "value": paasta_cluster},
+    ]
+    return extra_env
+
+
+def build_cell_config(cell, paasta_pool, paasta_cluster, region):
+    """
+    Build vtgate config
+    """
+    config = {
+        "name": cell,
+        "gateway": {
+            "extraFlags": {
+                "mysql_auth_server_impl": "vault",
+                "mysql_auth_vault_addr": f"https://vault-dre.{region}.yelpcorp.com:8200",
+                "mysql_auth_vault_path": "secrets/vitess/vt-gate/vttablet_credentials.json",
+                "mysql_auth_vault_tls_ca": f"/etc/vault/all_cas/acm-privateca-{region}.crt",
+                "mysql_auth_vault_ttl": "60s",
+            },
+            "affinity": build_affinity_spec(paasta_pool),
+            "extraLabels": build_extra_labels(paasta_pool, paasta_cluster),
+            "extraEnv": build_extra_env(paasta_cluster),
+            "replicas": 1,
+            "resources": {
+                "requests": {
+                    "cpu": "100m",
+                    "memory": "256Mi",
+                },
+                "limits": {"memory": "256Mi"},
+            },
+        },
+    }
+    vtgate_extra_env = copy.deepcopy(VTGATE_EXTRA_ENV)
+    for env in vtgate_extra_env:
+        if env["name"] == "VAULT_ADDR":
+            env["value"] = f"https://vault-dre.{region}.yelpcorp.com:8200"
+        if env["name"] == "VAULT_CACERT":
+            env["value"] = f"/etc/vault/all_cas/acm-privateca-{region}.crt"
+        config["gateway"]["extraEnv"].append(env)
+    return config
+
+
+def build_vitess_dashboard_config(cells, paasta_pool, paasta_cluster, zk_address):
+    """
+    Build vtctld config
+    """
+    config = {
+        "cells": cells,
+        "affinity": build_affinity_spec(paasta_pool),
+        "extraLabels": build_extra_labels(paasta_pool, paasta_cluster),
+        "extraEnv": build_extra_env(paasta_cluster),
+        "extraFlags": VTCTLD_EXTRA_FLAGS,
+        "replicas": 1,
+        "resources": {
+            "requests": {
+                "cpu": "100m",
```

Comment:
Same here, have to modify resource specifications based on service metadata in yelpsoa

Reply:
Fixed in https://github.com/Yelp/paasta/pull/3889/commits/6f56e5f7da82df3de44e035247fe865daf409ff4